packages/react/src/primitives/SwitchField/useSwitch.ts

```diff
@@ -17,7 +17,7 @@ export const useSwitch = (props) => {
         setIsOn(event.target.checked);
       }
     },
-    [onChange, isChecked]
+    [onChange, isControlled, isDisabled]
```

Comment:
Uneeded dependency `isChecked`, and missing `isControlled` and `isDisabled`

Reply:
good catch, eslint! :D